livre/eq/Channel.cpp

```diff
@@ -95,6 +93,37 @@ struct RedrawFilter : public Filter
 };
 
 
+struct SendHistogramFilter : public Filter
+{
+    explicit SendHistogramFilter( Channel* channel )
+        : _channel( channel )
+    {}
+
+    ~SendHistogramFilter() {}
+    void execute( const FutureMap& input,
+                  PromiseMap& ) const final
+    {
+        const UniqueFutureMap uniqueInputs( input.getFutures( ));
+        auto histogram = uniqueInputs.get< Histogram >( "Histogram" );
+        const auto& frameCounter =  uniqueInputs.get< uint32_t >( "Id" );
+        histogram.setId( frameCounter );
+        const_cast< eq::Config *>( _channel ->getConfig())->sendEvent( HISTOGRAM_DATA ) << histogram;
```

Comment:
This code makes me think that id should not be part of Histogram. It looks like the field is only meaningful inside the implementation of gatherHistogram. It may look convenient to pack the histogram and the frame number in a single object, but that's only correct if this pair is always well defined.
This means that you should either assign the frame id from the very beginning of send the two things separately in the HISTOGRAM_DATA event. The second option would change the implementation of gatherHistogram, but not significantly.